### DISCUSSION TITLE
what's the best way to dynamically update plugins

### DISCUSSION THREAD

**Discussion by zfan40** [katspaugh/wavesurfer.js#3751] [Q&A]
I am experimenting [wavesurfer react](https://www.npmjs.com/package/@wavesurfer/react) in combination with plugins. Most examples I found online were fixed plugin setups. 

My question is what's the best way to dynamically update plugins. it seems like ws is not receiving the plugin's updated config.

currently the code is like:

```javascript
// height is a variable
const plugins = () => [TimelinePlugin.create({
height: height,
})]
<WavesurferPlayer
plugins = {plugins}
/>
``` 
A bit of advice would be every helpful.

Thanks.

**Comment by zfan40** [comment]
I am not sure if this is similar to https://github.com/katspaugh/wavesurfer.js/discussions/3095
which means I need to destroy and re-create to update plugin settings

**Reply by katspaugh** [reply]
You can dynamically update plugins by destroying and registering them like this:

```js
timeline.destroy()

timeline = Timeline.create({ ... })

wavesurfer.registerPlugin(timeline)
```
